Implementing data model for Messages apps

Gal-f opened the pull request:
This commit includes all the neccessary files for the Messages app,
    including most importantly the data model.
Additionally modified 'settings.py' to familiarize the new app.

fix #42

Gal-f committed:
Implementing date model for Messages apps

This commit includes all the required files for the Messages app,
    including most importantly the data model.
Additionally modified 'settings.py' to familiarize the new app.

ravidbro commented on messages/admin.py:
```diff
@@ -0,0 +1,4 @@
+# from django.contrib import admin
+# Remove comment when implementing admin
+
+# Register your models here.
```
I think that you need to register your models

Gal-f replied:
fixed

ravidbro commented on messages/models.py:
```diff
@@ -0,0 +1,29 @@
+from django.db import models
```
I think that the migrations file should be in the same PR of the model that present them, it's generated from that

ravidbro commented on messages/models.py:
```diff
@@ -0,0 +1,29 @@
+    date = models.DateTimeField(default=timezone.now)
+    text = models.TextField()
+    # Message.id field auto-generated by Django, will be used down the road but no need to define it
+    category = models.TextField(max_length=50)
```
Should this be a free text or is there a table that defines categories?

Gal-f replied:
fixed

ravidbro commented on messages/models.py:
```diff
@@ -0,0 +1,29 @@
+
+
+class Message(models.Model):
+    user = models.ManyToManyField(User, blank=True)
```
I'm not sure it's doing what you meant, is the same message belongs to several users?

Gal-f replied:
fixed

ravidbro commented on messages/models.py:
```diff
@@ -0,0 +1,29 @@
+    @classmethod
+    def addMessage(cls, user, category, text):
+        newMessage = Message.objects.create()
+        newMessage.user = user
```
If the relation is manyToMany then 'user' is a list of users and not a single one.

Gal-f replied:
fixed

ravidbro commented:
Flake8 is complaining

ravidbro commented on messages/models.py:
```diff
@@ -0,0 +1,36 @@
+        newMessage.user = user
+        newMessage.category = category
+        newMessage.text = text
+        newMessage.date = timezone.now
```
no need if this is the default

Gal-f replied:
fixed

ravidbro commented on messages/models.py:
```diff
@@ -0,0 +1,36 @@
+
+class Category(models.Model):
+    name = models.TextField(max_length=50)
+    message = models.ManyToManyField(Message, blank=True)
```
it looks from Django documentation that ManyToMany is used only in one of the related classes:
https://docs.djangoproject.com/en/3.1/topics/db/examples/many_to_many/

Gal-f replied:
That's it!!
When both fields were listed, there was a mutual dependency so that even changing the order of classes in the file wouldn't help. Thanks!

Gal-f committed:
App name change - messages -> forumMessages

Due to conflicts, 'messages' app name was changed
to 'forumMessages'.
Updated in file names, folder, settings.py and admin.py.

Gal-f committed:
Fixing the data model for forumMessages

Changing class order to prevent unknown class errors,
fixing the double use of ManyToMany that defines the relation of
Message class-Category class.

This is a new commit since there are now 2 commits on the PR:
- Implementing the data model
- App name change
so commiting with 'amend' to the older commit is too tricky for me.

ravidbro commented on StuDash/settings.py:
```diff
@@ -29,6 +29,8 @@
     "main",
     "accounts",
+    "forumMessages",
+    "forumMessages.apps.ForumMessagesConfig",
```
Why do you need that line?

ravidbro replied:
I think you don't need this line, but I will merge it anyhow

Gal-f replied:
It was indeed unnecessary, even caused a duplication error in some scenarios.
I've fixed this on #49 .
